### PR TITLE
refactor!: bump libs version and new `-A` option behavior

### LINKED ISSUE #2371: [New Feature]: Include each syscall in Falco rulesets as default and new -A behavior
**Motivation**

Refactor Falco's syscalls activations mechanisms in order to solely activate the syscalls that are essential / necessary in order to "right-size/justify" CPU and memory utilizations (part 1). 

**Feature**

Current options:

At the moment (Falco 0.33.x) Falco activates a static set of system calls regardless of the syscalls defined in each Falco rule. If the end user chooses to trace and log a system call not defined in the static set, `-A` flag needs to be set and `-A` currently activates all syscalls Falco supports.

New desired behavior:
- Include all syscalls from each Falco rule automatically as new default behavior
- In addition, change `-A` flag behavior:
  - default (`-A` not set): all syscalls in rules included, sinsp state enforcement (Falco's static set) without high volume I/O syscalls
  - new `-A` flag: all syscalls in rules included, sinsp state enforcement AND allowing high volume I/O syscalls
- New falco-logger outputs at Falco startup including a sorted list of syscalls names that are activated 

**Additional context**

https://github.com/falcosecurity/falco/pull/2361

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind cleanup

/kind feature

/kind design

**Any specific area of the project related to this PR?**

/area build

/area engine

/area tests

**What this PR does / why we need it**:

This PR introduce the same changes of #2361 (the branch is actually based on that entirely for the first part), and adapts it by using the newest definitions and APIs introduced in libsinsp. The libs and driver versions have been bumped accordingly.

**Which issue(s) this PR fixes**:

Fixes #2371

**Special notes for your reviewer**:

We can decide to make this superseed #2361, or eventually merging #2361 first and rebase this on top of master after.
For the commits and changes picked from #2361, the git blame and co-authors have been preserved.

/milestone 0.35.0

cc @incertum 

**Does this PR introduce a user-facing change?**:

```release-note
refactor!: bump libs version and new `-A` option behavior
```

### CODE REVIEW

File: userspace/falco/app/actions/print_ignored_events.cpp [hunk 1]
```diff
@@ -16,51 +16,22 @@ limitations under the License.
 
 #include "actions.h"
 #include "helpers.h"
+#include "falco_utils.h"
```
Comment (incertum) [inline userspace/falco/app/actions/print_ignored_events.cpp:19]: seems we can remove "falco_utils.h" and "helpers.h" again

File: userspace/falco/app/actions/print_ignored_events.cpp [hunk 2]
```diff
@@ -16,51 +16,22 @@ limitations under the License.
...
 
 using namespace falco::app;
 using namespace falco::app::actions;
+using namespace falco::utils;
```
Comment (incertum) [inline userspace/falco/app/actions/print_ignored_events.cpp:23]: seems we can remove again

File: userspace/falco/app/actions/helpers_interesting_sets.cpp [hunk 3]
```diff
@@ -14,49 +14,137 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-#include "helpers.h"
+#include "actions.h"
+#include <sstream>
```
Comment (incertum) [inline userspace/falco/app/actions/helpers_interesting_sets.cpp:18]: seems we can remove again

File: userspace/falco/app/actions/helpers_interesting_sets.cpp [hunk 4]
```diff
@@ -14,49 +14,137 @@ See the License for the specific language governing permissions and
...
+	auto unsupported_event_set = rules_event_set.diff(s.selected_event_set);
+	if (unsupported_event_set.empty())
+	{
+		return;
+	}
+
+	/* Get the names of the events (syscall and non syscall events) that were not activated and print them. */
+	auto names = libsinsp::events::event_set_to_names(unsupported_event_set);
+	std::cerr << "Loaded rules match event types that are not activated or unsupported with current configuration: warning (unsupported-evttype): " + concat_set_in_order(names) << std::endl;
+	std::cerr << "If syscalls in rules include high volume I/O syscalls (-> activate via `-A` flag), else (2) syscalls might be associated with syscalls undefined on your architecture (https://marcin.juszkiewicz.com.pl/download/tables/syscalls.html)" << std::endl;
```
Comment (incertum) [inline userspace/falco/app/actions/helpers_interesting_sets.cpp:55]:
```
std::cerr << "If syscalls in rules include high volume I/O syscalls (-> activate via `-A` flag), else syscalls might be associated with syscalls undefined on your architecture (https://marcin.juszkiewicz.com.pl/download/tables/syscalls.html)" << std::endl;
```

File: userspace/falco/app/actions/helpers_interesting_sets.cpp [hunk 5]
```diff
@@ -14,49 +14,137 @@ See the License for the specific language governing permissions and
...
+static void select_event_set(falco::app::state& s, const libsinsp::events::set<ppm_event_code>& rules_event_set)
+{
+	/* PPM syscall codes (sc) can be viewed as condensed libsinsp lookup table
+	 * to map a system call name to it's actual system syscall id (as defined
+	 * by the Linux kernel). Hence here we don't need syscall enter and exit distinction. */
+	auto rules_names = libsinsp::events::event_set_to_names(rules_event_set);
+	if (!rules_event_set.empty())
+	{
+		falco_logger::log(LOG_DEBUG, "(" + std::to_string(rules_names.size())
+			+ ") syscalls activated in rules: " + concat_set_in_order(rules_names) + "\n");
```
Comment (incertum) [inline userspace/falco/app/actions/helpers_interesting_sets.cpp:67]:
```
+ ") events in rules: " + concat_set_in_order(rules_names) + "\n");
```

File: userspace/falco/app/actions/helpers_interesting_sets.cpp [hunk 6]
```diff
@@ -14,49 +14,137 @@ See the License for the specific language governing permissions and
...
+	 * -> Final set is union of (1) and (2) */
+	auto base_event_set = libsinsp::events::sinsp_state_event_set();
+	s.selected_event_set = rules_event_set.merge(base_event_set);
+
+	/* Derive the diff between the additional syscalls added via libsinsp state
+	enforcement and the syscalls from each Falco rule. */
+	auto non_rules_event_set = s.selected_event_set.diff(rules_event_set);
+	if (!non_rules_event_set.empty())
+	{
+		falco_logger::log(LOG_DEBUG, "+(" + std::to_string(non_rules_event_set.size())
```
Comment (incertum) [inline userspace/falco/app/actions/helpers_interesting_sets.cpp:84]: fix double counting
```
auto non_rules_event_set_names = libsinsp::events::event_set_to_names(non_rules_event_set);
falco_logger::log(LOG_DEBUG, "+(" + std::to_string(non_rules_event_set_names.size())
	+ ") events (includes Falco's state engine set of events): "
	+ concat_set_in_order(non_rules_event_set_names) + "\n");
```

File: userspace/falco/app/actions/helpers_interesting_sets.cpp [hunk 7]
```diff
@@ -14,49 +14,137 @@ See the License for the specific language governing permissions and
...
+		if (!erased_event_set.empty())
 		{
-			/* So far we only covered syscalls, so we add other kinds of
-			interesting events. In this case, we are also interested in
-			metaevents and in the procexit tracepoint event. */
-			if (sinsp::is_metaevent(ev) || ev == PPME_PROCEXIT_1_E)
-			{
-				s.ppm_event_info_of_interest.insert(ev);
-			}
+			falco_logger::log(LOG_DEBUG, "-(" + std::to_string(erased_event_set.size())
```
Comment (incertum) [inline userspace/falco/app/actions/helpers_interesting_sets.cpp:101]: same
```
auto erased_event_set_names = libsinsp::events::event_set_to_names(erased_event_set);
falco_logger::log(LOG_DEBUG, "-(" + std::to_string(erased_event_set_names.size())
    + ") ignored events (-> activate via `-A` flag): "
    + concat_set_in_order(erased_event_set_names) + "\n");
```

File: userspace/falco/app/actions/helpers_interesting_sets.cpp [hunk 8]
```diff
@@ -14,49 +14,137 @@ See the License for the specific language governing permissions and
...
 
-	/* In this case we get the tracepoints for the `libsinsp` state and we remove
-	 * the `sched_switch` tracepoint since it is highly noisy and not so useful
-	 * for our state/events enrichment.
-	 */
-	s.tp_of_interest = inspector->enforce_sinsp_state_tp();
-	s.tp_of_interest.erase(SCHED_SWITCH);
+	if (!s.selected_event_set.empty())
+	{
+		falco_logger::log(LOG_DEBUG, "(" + std::to_string(s.selected_event_set.size())
```
Comment (incertum) [inline userspace/falco/app/actions/helpers_interesting_sets.cpp:109]: same
```
auto selected_event_set_names = libsinsp::events::event_set_to_names(s.selected_event_set);
falco_logger::log(LOG_DEBUG, "(" + std::to_string(selected_event_set_names.size())
	+ ") events selected in total (final set): "
	+ concat_set_in_order(selected_event_set_names) + "\n");
```

**Review by incertum — DISMISSED** [review]
Awesome job merging in new changes Jason! On board with consistently dealing with events and also displaying events names in debug logs, left few comments around that!

**Comment by incertum** [comment]
please /hold until we have added more tests in libs 🙏 , thanks a bunch!

**Review by FedeDP — APPROVED** [review]
/approve

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**